### DISCUSSION TITLE
Using new @urql/svelte 2, cache-and-network doesn't seem to work properly

### DISCUSSION THREAD

**Discussion by frederikhors** [urql-graphql/urql#2460] [Q&A]
Trying for the first time the new @urql/svelte 2 package, transforming this:

```typescript
import { operationStore, query } from "@urql/svelte";

const playerStore = operationStore(
  playerDocument,
  undefined,
  { requestPolicy: "cache-and-network" }
);

query(playerStore);
```

to this:

```typescript
import { queryStore } from "@urql/svelte";

const playerStore = queryStore({
  client,
  query: playerDocument,
  variables: {},
  requestPolicy: "cache-and-network",
});
```

I found out that `cache-and-network` is not working properly: the request does not re-start when I reload the browser page.

`network-only` works.

@jonathanstanley, is it my fault?

**Comment by frederikhors** [comment]
@jonathanstanley is `cache-and-network` working to you?

**Reply by jonathanstanley** [reply]
I haven't been able to look at the updated PR. I know there were still some things I was trying to work through but maybe they were fixed. I will have this on my radar to circle back to, but afraid I'm time constrained for a little while.

**Reply by frederikhors** [reply]
I wrote to you because I thought that since you are the author you are using it in your projects right now. And hence `cache-and-network` is vital I was wondering. Take your time. :smile:

**Comment by freb** [comment]
Your code (well, my version of your code) is working for me. I have two pages in svelte-kit, and whenever I toggle back to the one where I explicitly set `requestPolicy: 'cache-and-network'`, a network request is made. I looked at the devtools as well and confirmed that of the two queryStores on the page, the one with my default `cache-first` has a cache-hit each time and does not submit a network request. The one that I changed to `cache-and-network` has a cache hit but still submits the network request.

**Comment by jonathanstanley** [comment]
I really appreciate urql and spent a good amount of time on this. I originally planned to provide documentation in addition to my code comments. However, I was getting stuck in some places.  I was waiting for more feedback so that I could make additional changes...but I was surprised by the merge. This also came right as I was realizing that urql may not work for our needs. for example: we need to distinguish network fetch from a cache "fetch". Since that is taking so much time to build, I'm not able to provide as much help as I had hoped here. 

If my PR is a problem, I totally understand if it is reverted. Or, if someone else can help write docs, I'm sure urql team would be happy to receive a PR for that. Apologies I can't be more help right now and for my delay in responding.
